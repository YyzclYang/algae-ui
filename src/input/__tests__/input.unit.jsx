import React from 'react';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Input from '../index';
import Button from '../../button';

describe('Input', () => {
  it('渲染一个 Input', () => {
    const component = renderer.create(<Input />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染 clear 按钮', () => {
    const component = mount(<Input allowClear value="v" onChange={() => {}} />);
    expect(component.find('svg')).toHaveLength(1);
  });

  it('点击 clear 按钮执行 clearFn', () => {
    const fn = jest.fn();
    const component = mount(
      <Input allowClear clearFn={fn} value="v" onChange={() => {}} />
    );
    component.find('svg').simulate('click');
    expect(fn).toBeCalled();
  });

  it('接受 inputAfterNode', () => {
    const node = <Button>node</Button>;
    const component = mount(
      <Input inputAfterNode={node} value="v" onChange={() => {}} />
    );
    expect(component.find('button')).toHaveLength(1);
  });
});
