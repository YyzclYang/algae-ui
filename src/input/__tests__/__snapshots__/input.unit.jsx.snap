// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input 渲染一个 Input 1`] = `
<span
  className="algae-ui-input-wrapper"
>
  <input
    className="algae-ui-input"
    style={Object {}}
    type="text"
  />
</span>
`;
